src/matrix.jl

```diff
@@ -392,7 +392,7 @@ function getsolution(m::RepMatrix, col::Int)
     if found
         out = convertoutput(output, Q.n, Q.hull == Clrs_true)
     else
-        out = nothing
+        out = Rational{BigInt}[]
```

Comment:
Why is this needed ?

Reply:
This is not really needed. It is to resolve the type instability of `getsolution`.

Reply:
I didn't know that. I just felt scared by the red color in the output of `@code_warntype`. I can revert the second commit if its change is inappropriate.

Reply:
I think it is better to revert it. See the following benchmark:
```
julia> f() = Rational{BigInt}[]
f (generic function with 1 metjulia> f() = Rational{BigInt}[]
f (generic function with 1 method)

julia> g() = nothing
g (generic function with 1 method)

julia> t(f) = @time f()
t (generic function with 1 method)

julia> h() = isempty(f())
h (generic function with 1 method)

julia> i() = g() === nothing
i (generic function with 1 method)

julia> t(h)
  0.000001 seconds (1 allocation: 80 bytes)
true

julia> t(h)
  0.000000 seconds (1 allocation: 80 bytes)
true

julia> t(i)
  0.000000 seconds
true

julia> t(i)
  0.000000 seconds
true

julia> using BenchmarkTools
[ Info: Recompiling stale cache file /home/blegat/.julia/compiled/v1.0/BenchmarkTools/ZXPQo.ji for BenchmarkTools [6e4b80f9-dd63-53aa-95a3-0cdb28fa8baf]

julia> @btime h()
  19.100 ns (1 allocation: 80 bytes)
true

julia> @btime i()
  0.025 ns (0 allocations: 0 bytes)
true

julia> @btime i()
  0.026 ns (0 allocations: 0 bytes)
truejulia> f() = Rational{BigInt}[]
f (generic function with 1 method)

julia> g() = nothing
g (generic function with 1 method)

julia> t(f) = @time f()
t (generic function with 1 method)

julia> h() = isempty(f())
h (generic function with 1 method)

julia> i() = g() === nothing
i (generic function with 1 method)

julia> t(h)
  0.000001 seconds (1 allocation: 80 bytes)
true

julia> t(h)
  0.000000 seconds (1 allocation: 80 bytes)
true

julia> t(i)
  0.000000 seconds
true

julia> t(i)
  0.000000 seconds
true

julia> using BenchmarkTools
[ Info: Recompiling stale cache file /home/blegat/.julia/compiled/v1.0/BenchmarkTools/ZXPQo.ji for BenchmarkTools [6e4b80f9-dd63-53aa-95a3-0cdb28fa8baf]

julia> @btime h()
  19.100 ns (1 allocation: 80 bytes)
true

julia> @btime i()
  0.025 ns (0 allocations: 0 bytes)
true

julia> @btime i()
  0.026 ns (0 allocations: 0 bytes)
truejulia> f() = Rational{BigInt}[]
f (generic function with 1 method)

julia> g() = nothing
g (generic function with 1 method)

julia> t(f) = @time f()
t (generic function with 1 method)

julia> h() = isempty(f())
h (generic function with 1 method)

julia> i() = g() === nothing
i (generic function with 1 method)

julia> t(h)
  0.000001 seconds (1 allocation: 80 bytes)
true

julia> t(h)
  0.000000 seconds (1 allocation: 80 bytes)
true

julia> t(i)
  0.000000 seconds
true

julia> t(i)
  0.000000 seconds
true

julia> using BenchmarkTools
[ Info: Recompiling stale cache file /home/blegat/.julia/compiled/v1.0/BenchmarkTools/ZXPQo.ji for BenchmarkTools [6e4b80f9-dd63-53aa-95a3-0cdb28fa8baf]

julia> @btime h()
  19.100 ns (1 allocation: 80 bytes)
true

julia> @btime i()
  0.025 ns (0 allocations: 0 bytes)
true

julia> @btime i()
  0.026 ns (0 allocations: 0 bytes)
truehod)
julia> f() = Rational{BigInt}[]
f (generic function with 1 method)

julia> g() = nothing
g (generic function with 1 method)

julia> t(f) = @time f()
t (generic function with 1 method)

julia> h() = isempty(f())
h (generic function with 1 method)

julia> i() = g() === nothing
i (generic function with 1 method)

julia> t(h)
  0.000001 seconds (1 allocation: 80 bytes)
true

julia> t(h)
  0.000000 seconds (1 allocation: 80 bytes)
true

julia> t(i)
  0.000000 seconds
true

julia> t(i)
  0.000000 seconds
true

julia> using BenchmarkTools
[ Info: Recompiling stale cache file /home/blegat/.julia/compiled/v1.0/BenchmarkTools/ZXPQo.ji for BenchmarkTools [6e4b80f9-dd63-53aa-95a3-0cdb28fa8baf]

julia> @btime h()
  19.100 ns (1 allocation: 80 bytes)
true

julia> @btime i()
  0.025 ns (0 allocations: 0 bytes)
true

julia> @btime i()
  0.026 ns (0 allocations: 0 bytes)
true
julia> g() = nothing
g (generic function with 1 method)

julia> t(f) = @time f()
t (generic function with 1 method)

julia> h() = isempty(f())
h (generic function with 1 method)

julia> i() = g() === nothing
i (generic function with 1 method)

julia> t(h)
  0.000001 seconds (1 allocation: 80 bytes)
true

julia> t(h)
  0.000000 seconds (1 allocation: 80 bytes)
true

julia> t(i)
  0.000000 seconds
true

julia> t(i)
  0.000000 seconds
true

julia> using BenchmarkTools
[ Info: Recompiling stale cache file /home/blegat/.julia/compiled/v1.0/BenchmarkTools/ZXPQo.ji for BenchmarkTools [6e4b80f9-dd63-53aa-95a3-0cdb28fa8baf]

julia> @btime h()
  19.100 ns (1 allocation: 80 bytes)
true

julia> @btime i()
  0.025 ns (0 allocations: 0 bytes)
true

julia> @btime i()
  0.026 ns (0 allocations: 0 bytes)
true
```

Reply:
Confirmed there is no significant difference:

```jl
f(flag) = flag ? [BigInt(1)//1] : Rational{BigInt}[]  # type stable
g(flag) = flag ? [BigInt(1)//1] : nothing  # type unsable

function F(flags)
    out = Vector{Rational{BigInt}}[]
    for i in eachindex(flags)
        a = f(flags[i])
        if !isempty(a)
            push!(out, a)
        end
    end
    out
end

function G(flags)
    out = Vector{Rational{BigInt}}[]
    for i in eachindex(flags)
        a = g(flags[i])
        if a !== nothing
            push!(out, a)
        end
    end
    out
end

flags = rand(Bool, 10^4);
```

```jl
@btime F(flags);
```

```
  3.982 ms (116525 allocations: 3.06 MiB)
```

```jl
@btime G(flags);
```

```
  3.804 ms (111597 allocations: 2.68 MiB)
```